test/index.spec.ts

```diff
@@ -60,22 +60,44 @@ describe('resin-semver', () => {
 			expect(semver.compare('2.0.0+rev2', 'Resin OS 2.0.0+rev5')).to.equal(-1);
 			expect(semver.compare('2.0.0+rev3', '2.0.0+rev3')).to.equal(0);
+			expect(semver.compare('2.0.0+rev1', '2.0.0+rev11')).to.equal(-1);
+			expect(semver.compare('2.0.0+rev2', '2.0.0+rev11')).to.equal(-1);
 		});
 
 		it('should correctly compare ".dev" versions', () => {
-			expect(semver.compare('2.0.0', '2.0.0+dev')).to.equal(1);
+			expect(semver.compare('2.0.0', '2.0.0+dev')).to.equal(-1);
 			expect(semver.compare('2.0.0', '2.0.0-dev')).to.equal(1);
-			expect(semver.compare('2.0.0', '2.0.0.dev')).to.equal(1);
-			expect(semver.compare('2.0.0+rev6', '2.0.0+rev6.dev')).to.equal(1);
-			expect(semver.compare('2.0.0+rev6.dev', '2.0.0+rev6')).to.equal(-1);
+			expect(semver.compare('2.0.0', '2.0.0.dev')).to.equal(1);// B is invalid
+			expect(semver.compare('2.0.0+rev6', '2.0.0+rev6.dev')).to.equal(-1);
+			expect(semver.compare('2.0.0+rev6.dev', '2.0.0+rev6')).to.equal(1);
 			expect(semver.compare('2.0.0+rev6.dev', '2.0.0+rev6.dev')).to.equal(0);
-			expect(semver.compare('Resin OS 2.0.0+rev3', 'Resin OS 2.0.0+rev3.dev')).to.equal(1);
-			expect(semver.compare('Resin OS 2.0.0+rev3.dev', 'Resin OS 2.0.0+rev3')).to.equal(-1);
+			expect(semver.compare('Resin OS 2.0.0+rev3', 'Resin OS 2.0.0+rev3.dev')).to.equal(-1);
+			expect(semver.compare('Resin OS 2.0.0+rev3.dev', 'Resin OS 2.0.0+rev3')).to.equal(1);
 			expect(semver.compare('Resin OS 2.0.0+rev3.dev', 'Resin OS 2.0.0+rev3.dev')).to.equal(0);
-			expect(semver.compare('2.0.0', 'Resin OS 2.0.0.dev')).to.equal(1);
-			expect(semver.compare('Resin OS 2.0.0.dev', '2.0.0')).to.equal(-1);
+			expect(semver.compare('2.0.0', 'Resin OS 2.0.0.dev')).to.equal(1);// B is invalid
+			expect(semver.compare('Resin OS 2.0.0.dev', '2.0.0')).to.equal(-1);// A is invalid
 			expect(semver.compare('Resin OS 2.0.0.dev', '2.0.0.dev')).to.equal(0);
 		});
+
+		it('should correctly compare "(dev)" and "(prod)" versions without revisions', () => {
+			expect(semver.compare('Resin OS 2.0.0', 'Resin OS 2.0.0 (dev)')).to.equal(-1);
+			expect(semver.compare('Resin OS 2.0.0', 'Resin OS 2.0.0 (prod)')).to.equal(-1);
+			expect(semver.compare('Resin OS 2.0.0 (dev)', 'Resin OS 2.0.0 (prod)')).to.equal(-1);
+
+			expect(semver.compare('Resin OS 2.0.0', 'Resin OS 2.0.0-prod')).to.equal(1);
+			expect(semver.compare('Resin OS 2.0.0', 'Resin OS 2.0.0.prod')).to.equal(1);// B is invalid
+			expect(semver.compare('Resin OS 2.0.0 (dev)', 'Resin OS 2.0.0+prod')).to.equal(-1);
+		});
+
+		it('should correctly compare "(dev)" and "(prod)" versions of the same revision', () => {
+			expect(semver.compare('Resin OS 2.0.0+rev3', 'Resin OS 2.0.0+rev3 (dev)')).to.equal(-1);
+			expect(semver.compare('Resin OS 2.0.0+rev3', 'Resin OS 2.0.0+rev3 (prod)')).to.equal(-1);
+			expect(semver.compare('Resin OS 2.0.0+rev3 (dev)', 'Resin OS 2.0.0+rev3 (prod)')).to.equal(-1);
+
+			expect(semver.compare('Resin OS 2.0.0+rev3', 'Resin OS 2.0.0+rev3-prod')).to.equal(-1);
```

Comment:
I don't mind this one, but I don't think it's necessary: this isn't a format we've ever used, and isn't really a good way of marking something as prod (build metadata should be separated with `.` only). If we wanted to remove this line from the test entirely, I'd be fine with that (and I think you might need to, given some of the other changes I've suggested).